### ISSUE TITLE
Language Support

### ISSUE THREAD

**Issue by Cir0X** [pjones/themoviedb#5]
TMDB API supports multiple [languages](https://developers.themoviedb.org/3/getting-started/languages) and it would be great if this lib would support languages, too.

**Comment by pjones** [comment]
@Cir0X That's a good idea, thanks for letting me know.  If you have the time I'd happy to merge a patch from you.

**Comment by Cir0X** [comment]
I'm pretty new in haskell, so I'm trying to find the best solution for this.

My first idea was to implement new localized functions, in order to provide backwards-compatibility.
Like this:

```haskell
fetchMovieLocalized :: ItemID -> Text -> TheMovieDB Movie
fetchMovieLocalized mid language = getAndParse ("movie/" ++ show mid) [("language", Just language)]
```

That would mean that for every function in Actions.hs there would be a xyzLocalized counterpart function:

* searchMoviesLocalized
* fetchMovieLocalized
* searchTVLocalized
* fetchTVLocalized
* fetchTVSeasonLocalized
* fetchFullTVSeriesLocalized

Furthermore would it be cleaner, when the second parameter is an type alias?

```haskell
type Language = Text
```

Alternatively there are packages wich support [ISO-639-1](https://en.wikipedia.org/wiki/ISO_639-1) and [ISO 3166-1](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2)

Or is there a better way?

**Comment by pjones** [comment]
Sorry for the delay in getting back to you on this.

I actually think it would be best if we added a language field to the `Configuration` type.  Then we would only have internal changes to make and the public API would not have to change.

Thoughts?

**Comment by Cir0X** [comment]
I've made a sketch for the language field, but I have some hard time figuring out what function should be applied to the field in the `FromJSON` implementation:

```haskell
data Configuration = Configuration
  { cfgImageBaseURL    :: Text   -- ^ The base URL for images.
  , cfgImageSecBaseURL :: Text   -- ^ Base URL for secure images.
  , cfgPosterSizes     :: [Text] -- ^ List of possible image sizes.
  , cfgLanguage        :: Text   -- ^ Specifies the language.
  } deriving (Generic)

--------------------------------------------------------------------------------
instance Binary Configuration

--------------------------------------------------------------------------------
instance FromJSON Configuration where
  parseJSON (Object v) =
    Configuration <$> images  "base_url"
                  <*> images  "secure_base_url"
                  <*> imagesM "poster_sizes" []
                  <*> ??????? "language" -- How the function would look like?
    where images key      = (v .: "images") >>= (.:  key)
          imagesM key def = (v .: "images") >>= (\x -> x .:? key .!= def)
  parseJSON _ = empty

--------------------------------------------------------------------------------
instance ToJSON Configuration where
  toJSON c = object [ "images" .= object
    [ "base_url"        .= cfgImageBaseURL c
    , "secure_base_url" .= cfgImageSecBaseURL c
    , "poster_sizes"    .= cfgPosterSizes c
    , "language"        .= cfgLanguage c
    ]]
```

**Comment by pjones** [comment]
@Cir0X Thanks so much for your effort on this!

The `FromJSON` instance should look like this:

```haskell
instance FromJSON Configuration where
  parseJSON (Object v) =
    Configuration <$> images  "base_url"
                  <*> images  "secure_base_url"
                  <*> imagesM "poster_sizes" []
                  <*> v .:? "language" .!= "en"
    where images key      = (v .: "images") >>= (.:  key)
          imagesM key def = (v .: "images") >>= (\x -> x .:? key .!= def)
  parseJSON _ = empty
```

This way if the `language` setting is missing from the YAML file it will default to `en`.  (I assume the default should be `en`.  If I'm wrong please adjust accordingly.)

Those funny operators come from the [aeson accessors docs](https://hackage.haskell.org/package/aeson-1.2.4.0/docs/Data-Aeson.html#g:19).

**Comment by Cir0X** [comment]
@pjones Thanks for your reply!
So the `.:?` function just checks if the `language` key is set and when it's not, it returns `Nothing`. So `Nothing  .!= "en"` returns `"en"`. Hypothetically if we want the `language` field to be mandatory, we could use `.:` instead.

I've applied your solution and all tests passed.
What I don't understand is: How can I configure e.g. the example `tmdb` executable to use a specific language now?
You mentioned that I can configure it in some YAML file, but I can't trace it in the source code. Can you give me a hint?

**Comment by pjones** [comment]
Your understanding of `.:?` and `.!:` is correct.

I misspoke about a YAML configuration file.  I was thinking about one of the tools I wrote that use this library.  So the `tmdb` executable will need to be changed so it has an additional command line argument for the language field.

**Comment by pjones** [comment]
@Cir0X Just in case you didn't notice, I released version 1.2 with support for selecting a language code.

Thank you for your work on this and I'm sorry I let it get so stale.
